### DISCUSSION TITLE
Document & add endDate handing to the UI

### DISCUSSION THREAD

**Discussion by jkamenik** [davish/obsidian-full-calendar#384] [Feature Requests]
No where in the docs is `endDate` described.  Nor is it in the UI.  However, it works.  The following two events both render according to spec.

```
---
title: Multi-day Timed event
allDay: false
startTime: 12:00
endTime: 12:30
date: 2023-01-22
endDate: 2023-01-24
completed: null
---
```

```
---
title: Multi-day All Day event
allDay: true
date: 2023-01-22
endDate: 2023-01-24
completed: null
---
```

## Thoughts on Non-inclusive end-dates

End dates/times are exclusive, while start dates/times are inclusive.  This behavior is noted in the [event parsing](https://fullcalendar.io/docs/event-parsing) section of Full Calendar doc.  And it is a also the iCal spec.  And it makes sense for computers, but is not intuitive.

It is done this way so that an event's end date can be the same as the next event's start date and regardless of system precision there is no overlap.  So, the first event ends at 12:29:59 on the 24th; which renders the way one expects  However, the second ends 11:59:59 on the 23rd and doesn't render on the 24th.

The later behavior seems like a bug to most humans.  Which is why most calendar apps handle this by making the event inclusive in the UI (i.e., from the 22 to the 24th) but generate the event exclusive (i.e., `endDate: 2023-01-25`) in the document.

Since we easily see the document first and foremost, it would likely be raised as a bug if we follow spec and made `endDate` exclusive since folks would have to remember it was exclusive.

One through might be to add and use `endDateInclusive` and for `allDay: true` events add 1 day before sending it for rendering.  Or add `endDateExclusive` as an alternative property to `endDate`.  For `allDay: false` still write out `endDate` since it makes sense.  And for `allDay: true` use `endDateExclusive` instead so it is more clear in the file what is happening.

Just some thoughts.

**Comment by xojojo** [comment]
I would greatly appreciate the addition of some way to mark end dates as inclusive in events for purposes of the graphical view. I use the metadata endDate to surface events that are ending on a given day, and having to work around the "manually add +1 day to show as expected in graphical view" exclusive-date workaround is very redundant. 

My other option is setting specific start and end times, but then I have timestamps showing in graphical view for events that don't need timestamps because they're actually all-day. A property to set inclusivity would help make the multi-day behaviour more intuitive.
